lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -305,11 +305,18 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 			selectedCoins, localContributionAmt, changeAmt,
 				err = CoinSelectUpToAmount(
 				r.FeeRate, r.MinFundAmt, r.FundUpToMaxAmt,
-				w.cfg.DustLimit, coins,
+				r.WalletReserve, w.cfg.DustLimit, coins,
 			)
 			if err != nil {
 				return err
 			}
+			// Ensure that the remote channel reserve does not
+			// exceed 20% of the channel capacity.
```

Comment:
nit: see my comment above I don't think we need this check here its already done in a higher level